document/CoreFormat.mdk

```diff
@@ -364,17 +448,57 @@ This allows tools to include information about the run that is not explicitly sp
 
 ### General { #fileReference-General }
 
-A `fileReference` object represents a single file.
+A `fileReference` object represents a single file. In some cases, this file might be nested within another file,
+referred to as the "containing file."
 
-### `uri` property { #fileReference-uri }
+### `pathFromParent` property { #fileReference-pathFromParent }
 
-A `fileReference` object shall contain a property named `uri` whose value is a string that represents
-the location of the file as a valid URI.
+If this `fileReference` object refers to a file which is nested within another file, and
+if the location of the nested file with respect to its containing file can be expressed
+by means of a path (as is the case for compressed containers such as Zip archives),
+then the `fileReference` object shall contain a property named `pathFromParent`
+whose value is a string specifying the path from the root
+of the containing file to the file represented by this `fileReference` object.
+
+If this `fileReference` object refers to a file which is not nested within another file,
+or if the location of the file within its containing file cannot be expressed by means of a path,
+then the `pathFromParent` property shall be absent. 
+
+### `offsetFromParent` property { #fileReference-offsetFromParent }
+
+If this `fileReference` object refers to a file which is nested within another file, and
+if the location of the nested file with respect to its containing file cannot be expressed
```

Comment:
i don't think we need to make pathFromParent and offsetFromParent mutually exclusive. you can imagine that a stream within a storage container might be referenceable by path but also can be loaded via a direct offset. IOW, i think these cases are valid, pathFromParent, pathFromParent + offsetFromParent, offsetFromParent.